### DISCUSSION TITLE
Changing authentication parameters from within .htaccess files

### DISCUSSION THREAD

**Discussion by mgkuhn** [OpenIDC/mod_auth_openidc#1270] [Q&A]
I would like to allow users of a multi-user web server to control mod_auth_openidc  parameters from within their respective personal homepage directories, via directives that they can add to their personal `.htaccess` files.

Would it be possible to use some of the [OIDC settings](https://github.com/OpenIDC/mod_auth_openidc/blob/master/auth_openidc.conf) within a .htaccess file? I've tried adding some `OIDC...` directives related to session expiry times to a .htaccess file, such as

```
OIDCSessionInactivityTimeout 300
OIDCSessionMaxDuration 3600
OIDCAuthRequestParams "max_age=3600"
```

but sadly they all resulted in  `500 Internal Server Error`.

**Comment by zandbelt** [comment]
no that is not possible: they are global settings, at least the first 2 are

**Comment by mgkuhn** [comment]
Are there at least some local settings that you could make configurable in a future release via .htaccess, for example `OIDCAuthRequestParams`?

We previously used [AuthType Ucam-WebAuth](https://github.com/cambridgeuniversity/mod_ucam_webauth) where [“All directives can appear in [...] `.htaccess` files if `AllowOverride AuthConfig` is in effect.”](https://github.com/cambridgeuniversity/mod_ucam_webauth/blob/6a246d5299171713c8680ef37c6ea6545bac948a/README.Config#L227)
